orm/serial_bucket.go

```diff
@@ -258,7 +258,10 @@ func (smb *serialModelBucket) IndexScan(db weave.ReadOnlyKVStore, indexName stri
 }
 
 func (smb *serialModelBucket) ByIndex(db weave.ReadOnlyKVStore, indexName string, key []byte, destination SerialModelSlicePtr) error {
-	// TODO refactor this function to return an iterator
+	// TODO Refactor this function to return lazy iterator
+	// get indexed loads all the objects to objs where
+	// the iterator wrapping must happen. Should be taken care
+	// serial model bucket is optimized by moving bucket logic here
```

Comment:
I change my mind, sorry for that.

Right now, the function loads all values and returns all of them at once. It returns only an error. I think this comment should be removed completely, because described change will not happen. If this bucket implementation will be used, it is easier to introduce another method instead and maybe remove this one.

Changing notation of a method is hard if you use composition. You cannot use whatever notation you like, you must use what is expected.

Reply:
> Changing notation of a method is hard if you use composition. You cannot use whatever notation you like, you must use what is expected.

Can you explain the `composition here`

Reply:
In Go, preffered is [composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance).

As an example, `migration` package does not inherit from `ModelBucket`. Instead, it enrich `ModelBucket` functionality by providing the very same interface with a custom implementation.
The above would not be possible if the notation of the functions would change. You would have to update all places where it is used.